python/histone_utility.py

```diff
@@ -0,0 +1,342 @@
+        df: dataframe of standard indexing, with columns being celllines and 
+            corresponding tissue (for the data used in the code, the tissue 
+            and cellline is seperated by an underscore)
+    
+    OUTPUT
+        tissue_dict: dictionary of cell line and corresponding tissue
+        cellline_list: list of cell lines in the dataframe
+        
+    """
+    celllines = list(df)
+    celllines.pop(0)
+
+    cellline_list =[]
+    tissue_list = []
+    for cell in celllines:
+        temp = cell.split('_')
+        cell = temp[0]
+        cellline_list.append(cell)
+        tissue = temp[1:]
+        tissue = " ".join(tissue)
+        tissue_list.append(tissue)
+
+    d = {'Celllines': cellline_list, 'Tissue' : tissue_list}
+    cell_df = pd.DataFrame(d)
+    cellline_tissue = cell_df.values
+    tissue_dict ={}
+    for data in cellline_tissue:
+        tissue_dict[data[0]] = data[1]
+    return tissue_dict, cellline_list
+
+
+def normalize(df, delete_column, cell_list):
+    """Takes in a dataframe with columns being cell lines 
+    (with the exception of a column being genes or gene_ids) and normalizes 
+    the expression values.
+    
+    This will delete the desired column and then normalize the values and 
+    fill in the NaN values. It will then format the dataframe to back 
+    to how it originally looked. 
+    
+    INPUT
+    
+        df: dataframe with columns of cell lines (with the exception of a 
+            column being genes or gene_ids)
+        delete_column: column that does not contain expression values
+        cell_list: list of cell lines within dataframe
+    
+    OUTPUT
+        df_normalized: dataframe of normalized expression values and 
+                       NaN values filled   
+                       
+    """
+
+    temp_df = df.copy()
+    
+    delete = temp_df[delete_column]
+    del temp_df[delete_column]
+    
+    x = temp_df.values.astype(float)
+    min_max_scaler = preprocessing.MinMaxScaler()
+    x_scaled = min_max_scaler.fit_transform(x)
+    df_normalized = pd.DataFrame(x_scaled)
+
+    df_normalized.insert(0, 'Gene', delete , True)
+    df_normalized.columns = ['Gene'] + cell_list
+    
+
+    # Fill in NaN values.
+    df_normalized = df_normalized.fillna(method='ffill')
+    return df_normalized
+
+
+def common(a,b):
+    """ This functions purpose is to find common elements of 
+    lists by converting each list to a set.
+    
+    The inputs of this function should both be lists. 
+    
+    INPUT
+        a: list which you wish to find common elements
+        b: list which you wish to find common elements
+        
+    OUTPUT
+        common: list containing common elements from a and b
+        
+    """
+    a_set = set(a)
+    b_set = set(b)
+    common = a_set.intersection(b_set) 
+    common = list(common)
+    return common 
+    
+def pearson_dfs(gene_list, histone_list, data1, data2, dtype):
+    """This functions purpose is to create two dataframes. 
```

Comment:
the purpose is not to create two dataframes: it is to compute the column-wise pearson correlation coefficient and p-value for each element in an array.